src/Asset/Library.php

```diff
@@ -295,196 +305,133 @@ public function buildByExtension($extension_type, $extension): array {
-        // Process libraries overrides.
-        if (isset($libraries_overrides["$extension/$library_name"])) {
-          // Active theme defines an override for this library.
-          $override_definition = $libraries_overrides["$extension/$library_name"];
-          if (is_string($override_definition) || $override_definition === FALSE) {
-            // A string or boolean definition implies an override (or removal)
-            // for the whole library. Use the override key to specify that this
-            // library will be overridden when it is called.
-            // @see \Drupal\Core\Asset\LibraryDiscovery::getLibraryByName()
-            if ($override_definition) {
-              $libraries[$library_name]['override'] = $override_definition;
-            }
-            else {
-              $libraries[$library_name]['override'] = FALSE;
-            }
-          }
-          elseif (is_array($override_definition)) {
-            // An array definition implies an override for an asset within this
-            // library.
-            foreach ($override_definition as $sub_key => $value) {
-              // Throw an exception if the asset is not properly specified.
-              if (!is_array($value)) {
-                throw new InvalidLibrariesOverrideSpecificationException(sprintf('Library asset %s is not correctly specified. It should be in the form "extension/library_name/sub_key/path/to/asset.js".', "$extension/$library_name/$sub_key"));
-              }
-              if ($sub_key === 'drupalSettings') {
-                // drupalSettings may not be overridden.
-                throw new InvalidLibrariesOverrideSpecificationException(sprintf('drupalSettings may not be overridden in libraries-override. Trying to override %s. Use hook_library_info_alter() instead.', "$extension/$library_name/$sub_key"));
-              }
-              if ($sub_key === 'css') {
-                // SMACSS category should be incorporated into the asset name.
-                foreach ($value as $category => $overrides) {
-                  $this->setOverrideValue($libraries[$library_name], [$sub_key, $category], $overrides, $theme_path);
-                }
-              }
-              else {
-                $this->setOverrideValue($libraries[$library_name], [$sub_key], $value, $theme_path);
-              }
-            }
-          }
-        }
+  public function getLibrariesByExtension($extension) {
+    if (!isset($this->libraryDefinitions[$extension])) {
+      $libraries = $this->get($extension);
+      $this->libraryDefinitions[$extension] = [];
+      foreach ($libraries as $name => $definition) {
+        $this->libraryDefinitions[$extension][$name] = $definition;
       }
     }
 
-    return $libraries;
+    return $this->libraryDefinitions[$extension];
   }
 
   /**
-   * Utility function to get all Patternkit module metadata.
+   * {@inheritDoc}
+   */
+  public function getLibraryAsset($key) {
```

Comment:
It seems that this method will try to fetch the `atoms/example/ref/src/example/ref` library (i.e. `substr($key, strlen($library_name . '/')) === 'atoms/example/ref/src/example/ref'`  in the current state of this branch, but I believe we want it to fetch `atoms/example/src/example`

Reply:
I'm actually not sure what real world scenario we're looking at here - if that's related to an internal library that using /ref/ paths, then this is the wrong place to address it and instead we'd want to add a plugin to handle retrieval.

Or am I reading this wrong?